### PR TITLE
Add and remove assignees

### LINKED ISSUE #560: Assign issues and pull requests

### PR DESCRIPTION
**Pull request by rnystrom**
Fixes #560 

img_0075

img_0076

### CODE REVIEW

File: Classes/Issues/Assignees/IssueAssigneeSummaryModel.swift [hunk 1]
```diff
@@ -14,11 +14,13 @@ final class IssueAssigneeSummaryModel: ListDiffable {
     let title: String
     let urls: [URL]
     let expanded: Bool
+    private let urlHash: String
```
Comment (rnystrom) [inline Classes/Issues/Assignees/IssueAssigneeSummaryModel.swift]: Necessary to get the summary nested view to update

File: Classes/Issues/GithubClient+Issues.swift [hunk 2]
```diff
@@ -389,4 +389,87 @@ extension GithubClient {
...
+            request(GithubClient.Request(
+                path: path,
+                method: .post,
+                parameters: ["assignees": added]
+            ) { (response, _) in
+                handler(201, response.response?.statusCode)
+            })
+        }
+
+        // https://developer.github.com/v3/issues/assignees/#remove-assignees-from-an-issue
```
Comment (rnystrom) [inline Classes/Issues/GithubClient+Issues.swift]: These endpoints are lame. Unlike labelling, adding and removing are in **separate** endpoints. There's definitely the case where one of these fails and the other succeeds, then you reset the data back to the _original_ state (not including success or failure), but I don't really care at this point. Should be rare.

**Comment by rnystrom** [comment]
Also in case you're curious about the iconography between "Assignees" and "Reviewers", I whipped up this icon tonight for reviewers:

reviewer 3x

File: Classes/Issues/GithubClient+Issues.swift (hunk 2 above)
Comment (Sherlouk) [reply Classes/Issues/GithubClient+Issues.swift]: That is such a stupid limitation, what...

File: Classes/People/PeopleViewController.swift [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+    func configure(
+        selections: [String],
+        type: PeopleType,
+        client: GithubClient,
+        delegate: PeopleViewControllerDelegate,
+        owner: String,
+        repo: String
+        ) {
+        self.selections = Set<String>(selections)
```
Comment (Sherlouk) [inline Classes/People/PeopleViewController.swift]: Not sure where the best place to make the change would be (maybe an option could get passed through to the `configure` method -- but we should always have the signed-in user as the first option to quickly assign yourself!

File: Classes/People/PeopleViewController.swift [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    private var users = [User]()
+    private let feedRefresh = FeedRefresh()
+    private var type: PeopleType!
+    private var selections = Set<String>()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        tableView.refreshControl = feedRefresh.refreshControl
+        feedRefresh.refreshControl.addTarget(self, action: #selector(LabelsViewController.onRefresh), for: .valueChanged)
```
Comment (Sherlouk) [inline Classes/People/PeopleViewController.swift]: `LabelsViewController` here. Fun fact you don't need to specify the class `#selector(onRefresh)` is just as valid :)

File: Classes/People/PeopleViewController.swift [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        if let cell = cell as? PeopleCell, let avatarURL = URL(string: user.avatar_url) {
+            let login = user.login
+            cell.configure(avatarURL: avatarURL, username: login, showCheckmark: selections.contains(login))
+        }
+        return cell
+    }
+
+    // MARK: UITableViewDelegate
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
```
Comment (Sherlouk) [inline Classes/People/PeopleViewController.swift]: https://developer.apple.com/documentation/uikit/uitableviewdelegate/1614943-tableview

We could implement `willSelectRowAt` and return `nil` if the user would exceed their selection count (preventing the visual selecting of the cell?

Another alternative would be to do a quick error modal saying "Exceeded Selection Limit"?

Comment (rnystrom) [reply Classes/People/PeopleViewController.swift]: I like the `willSelectRowAt` idea, that'll be clean. The error message might be a bit much, hoping people can pick up on the "3/10" label in the top left. Maybe we can do a scale "pop" animation on that label to draw their attention? IMO details we can add later.

File: Classes/People/PeopleViewController.swift (hunk 4 above)
Comment (rnystrom) [reply Classes/People/PeopleViewController.swift]: TIL

File: Classes/People/PeopleViewController.swift (hunk 3 above)
Comment (rnystrom) [reply Classes/People/PeopleViewController.swift]: GREAT idea. I'll look into adding a "Claim" button at the very top

**Comment by rnystrom** [comment]
Btw should have mentioned, I plan on reusing this for adding Reviewers to PRs since its almost exactly the same (notice the `PeopleType` enum).

**Comment by rnystrom** [comment]
Going to add "claim" as a followup blocked by #1026. Will make these much easier to manage.

**Pull request merged**